src/Analysis/Ast/Impl/Documents/DocumentBuffer.cs

```diff
@@ -17,27 +17,27 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
-using System.Threading;
-using Microsoft.Python.Core.Diagnostics;
 using Microsoft.Python.Parsing;
 
 namespace Microsoft.Python.Analysis.Documents {
     internal sealed class DocumentBuffer {
         private readonly StringBuilder _sb = new StringBuilder();
-        private readonly int _ownerThreadId = Thread.CurrentThread.ManagedThreadId;
+        private string _cachedText;
 
         public int Version { get; private set; }
-        public string Text => _sb.ToString();
+        public string Text => _cachedText ?? (_cachedText = _sb.ToString());
 
         public void Reset(int version, string content) {
             Version = version;
             _sb.Clear();
             if (!string.IsNullOrEmpty(content)) {
                 _sb.Append(content);
             }
+            _cachedText = null;
         }
 
         public void Update(IEnumerable<DocumentChange> changes) {
+
```

Comment:
`SplitLines` can also be optimized, cause it always calls `sb.ToString()`, but I'm not sure if that would be a huge improvement.

Reply:
Yes, I can just borrow https://github.com/Microsoft/RTVS/tree/master/src/VsCode/LanguageServer/Impl/Text too.  But it is not visible in the updated profile so it is low pri.